extensions/apps/extensions/src/components/pages/my-extensions/extension-element.tsx

```diff
@@ -0,0 +1,251 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import getSDK from '@akashaorg/awf-sdk';
+
+import Stack from '@akashaorg/design-system-core/lib/components/Stack';
+import AppAvatar from '@akashaorg/design-system-core/lib/components/AppAvatar';
+import Icon from '@akashaorg/design-system-core/lib/components/Icon';
+import Divider from '@akashaorg/design-system-core/lib/components/Divider';
+import Text from '@akashaorg/design-system-core/lib/components/Text';
+import Menu from '@akashaorg/design-system-core/lib/components/Menu';
+import {
+  Plugin,
+  App,
+  Widget,
+} from '@akashaorg/design-system-core/lib/components/Icon/akasha-icons';
+import {
+  ArrowUpOnSquareIcon,
+  ClockIcon,
+  EyeIcon,
+  PencilIcon,
+  XMarkIcon,
+} from '@heroicons/react/24/outline';
+import { MenuProps } from '@akashaorg/design-system-core/lib/components/Menu';
+import { EllipsisHorizontalIcon } from '@akashaorg/design-system-core/lib/components/Icon/hero-icons-outline';
+
+import { hasOwn, useRootComponentProps } from '@akashaorg/ui-awf-hooks';
+import { useGetAppsStreamQuery } from '@akashaorg/ui-awf-hooks/lib/generated/apollo';
+import {
+  AkashaAppApplicationType,
+  AkashaAppsStreamModerationStatus,
+  AppImageSource,
+} from '@akashaorg/typings/lib/sdk/graphql-types-new';
+import { ExtensionStatus } from '@akashaorg/typings/lib/ui';
+
+type Extension = {
+  id?: string;
+  name?: string;
+  displayName?: string;
+  logoImage?: AppImageSource;
+  description?: string;
+  applicationType?: AkashaAppApplicationType;
+};
+
+type ExtensionElement = {
+  extensionData: Extension;
+  showDivider?: boolean;
+  filter: { id?: string; title?: string; opt?: string };
+};
+
+export const ExtensionElement: React.FC<ExtensionElement> = ({
+  extensionData,
+  showDivider,
+  filter,
+}) => {
+  const { t } = useTranslation('app-extensions');
+  const sdk = React.useRef(getSDK());
+
+  const { navigateToModal } = useRootComponentProps();
+
+  const { data: appStreamReq } = useGetAppsStreamQuery({
+    variables: {
+      indexer: sdk.current.services.gql.indexingDID,
+      first: 1,
+      filters: {
+        where: {
+          applicationID: {
+            equalTo: extensionData.id,
```

Comment:
ensure that the id is not empty

Reply:
on line 72 there is the option passed to skip the query if id is nullish

Reply:
a string like `" "` would pass the validation, we need more strict checks for filters